imap/imapd.c

```diff
@@ -141,6 +141,13 @@ static int nosaslpasswdcheck = 0;
 static int apns_enabled = 0;
 static int64_t maxsize = 0;
 
+const struct client_behavior_registry client_behavior_null = {
+  .did_notify = 0,
+  .did_compress = 0,
+};
+
+struct client_behavior_registry client_behavior = client_behavior_null;
+
```

Comment:
You don't need `client_behaviour_null`

`client_behaviour` should be `static` (it's not used outside this module), and you can initialise it without `client_behaviour_null` here in a few ways:

```
static struct client_behaviour_registry client_behaviour;  // 'static' globals are implicitly initialized with zeroes
static struct client_behaviour_registry client_behaviour = { 0 };   // or do it semi-explicitly
static struct client_behaviour_registry client_behaviour = { 0, 0 };   // or really explicitly
```

Reply:
Amended, thank you very much!

Reply:
Hilariously, you wrote `behaviour` and I just copied that, so the code then had mismatched spellings.  I fixed it quickly, but it definitely made me smile. 😄